src/Grids/vertically_stretched_cartesian_grid.jl

```diff
@@ -149,3 +157,22 @@ function generate_stretched_vertical_grid(FT, z_topo, Nz, Hz, zF_generator)
 
     return Lz, zF, zC, ΔzF, ΔzC
 end
+
+# We cannot reconstruct a VerticallyStretchedCartesianGrid without the zF_generator.
+# So the best we can do is tell the user what they should have done.
```

Comment:
Should we ask users to provide only the interior grid, and pad the interior grid with halo points according to the size of `halo`?

Reply:
I'm a bit puzzeld by this change.  What is `zF_generator`?

If halo is set by what we need for the advection scheme, then I think we would need the interor halos to be as wide as the exterior halos.  Unless there is something that I'm missing?

Reply:
Ah the `zF_generator` is either an array or function of `k` passed to `VerticallyStretchedCartesianGrid` that determines the locations of the faces in the z-direction.

The problem is that `with_halo(new_halo, old_grid)` is called in the model constructor after the grid has been created so the grid cannot be reconstructed via simple call to `VerticallyStretchedCartesianGrid` as the `zF_generator` is not stored in the grid.

We could probably get around this but I just decided to print a warning for now as I wanted to focus on getting tests to pass.

Reply:
Ahh, I understand. Do we allow users to pass `zF` as an Array? In that case it might work to pass `grid.zF[1:Nz+1]` to the constructor for `VerticallyStretchedCartesianGrid`.